Add FAQ entry for "server sends HTTP headers in a loop"

kaduk opened the pull request:
Older versions of a few commercial HTTPS servers don't handle
extended-master-secret and/or encrypt-then-mac very well, but we end up
getting asked about this weird behavior that shows up when people
upgrade to OpenSSL 1.1.0 clients.

Text largely taken from the discussion at
https://github.com/openssl/openssl/issues/9360 .

kaduk committed:
Add FAQ entry for "server sends HTTP headers in a loop"

Older versions of a few commercial HTTPS servers don't handle
extended-master-secret and/or encrypt-then-mac very well, but we end up
getting asked about this weird behavior that shows up when people
upgrade to OpenSSL 1.1.0 clients.

Text largely taken from the discussion at
https://github.com/openssl/openssl/issues/9360 .

mspncp commented on docs/faq-2-user.txt:
```diff
@@ -213,3 +213,18 @@
+    encrypt-then-mac and extended-master-secret, both of which provide
+    significant security improvements.  Unfortunately, some deployed TLS
+    servers are fatally buggy and do not implement extensibility in a
+    standards-complaint manner; these servers may exhibit strange behavior
```
complaint -> compliant

mspncp commented on docs/faq-2-user.txt:
```diff
@@ -213,3 +213,18 @@
+    OpenSSL 1.1.0 introduced support for several new TLS extensions, including
+    encrypt-then-mac and extended-master-secret, both of which provide
+    significant security improvements.  Unfortunately, some deployed TLS
+    servers are fatally buggy and do not implement extensibility in a
```
I‘d remove „are fatally buggy and“. It sounds very biased.

t8m replied:
Well, breaking on unknown extensions is a quite fatal bug. Maybe use "are severally broken"?

mspncp replied:
+1 from me for @t8m's suggestion, it sounds less colloquial and more objective.

mspncp commented on docs/faq-2-user.txt:
```diff
@@ -213,3 +213,18 @@
+    servers are fatally buggy and do not implement extensibility in a
+    standards-complaint manner; these servers may exhibit strange behavior
+    such as repeating the HTTP headers in a loop after receiving a ClientHello
+    that includes "newer" TLS extensions such as these.  While these new TLS
```
Don’t place „newer“ in double quotes, it gives the word a weird emphasis. Also, I would use unknown instead of newer.

t8m replied:
... includes such TLS extensions unknown to them.

mspncp commented on docs/faq-2-user.txt:
```diff
@@ -213,3 +213,18 @@
+    standards-complaint manner; these servers may exhibit strange behavior
+    such as repeating the HTTP headers in a loop after receiving a ClientHello
+    that includes "newer" TLS extensions such as these.  While these new TLS
+    extensions provide significant security benefits to clients and are
```
Remove „and are rightly ... in OpenSSL“

t8m replied:
Why? Maybe replace "in OpenSSL" with "in contemporary TLS clients"?

mspncp reviewed (commented):
I weakened two sentences, because they sounded a bit lecturing in my ears. If you think differently about it, feel free to ignore my suggestions.

kaduk committed:
fixup! Add FAQ entry for "server sends HTTP headers in a loop"

kaduk commented:
> I weakened two sentences, because they sounded a bit lecturing in my ears. If you think differently about it, feel free to ignore my suggestions.

My main goal was to be able to close https://github.com/openssl/openssl/issues/9360 so I'm happy to make changes here.
I pushed an update that hopefully addresses the comments properly  Thanks for the review!

kaduk commented:
I forget; do I need someone OTC-shaped to actually merge things in this repo?

mspncp reviewed (approved):
LGTM

mspncp commented:
> I forget; do I need someone OTC-shaped to actually merge things in this repo?

I think this repository is still owned by the @openssl/omc. The handover hasn't happened yet.

mattcaswell commented:
> I think this repository is still owned by the @openssl/omc. The handover hasn't happened yet.

It is still OMC owned and  requires 2 OMC approvals at the moment. So far it has not been proposed to transfer control of web to OTC - only tools. The OMC have approved the transfer of ownership of the tools repo.

mattcaswell commented:
Needs second OMC approval...

kaduk commented:
It looks like this got all the approvals it needs, but needs a volunteer to actually merge it, if I remember correctly the way the permissions are set up.

mattcaswell commented:
Pushed.
